### PR TITLE
feat: monitoring support for k8s

### PR DESCRIPTION
**Pull request by samsja**

### CODE REVIEW

**Comment by samsja** [comment]
Don't merge this pr for now I am preparing the k8s documentation on monitoring in a seperate PR dedicated to monitoring documentation. I want to be sure that what I am saying in the documentation on k8s is working before merging this one.

So I will merge both of them when they are ready

File: jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def get_deployment_yamls(
     container_cmd_uses_after: Optional[str] = None,
     container_args_uses_before: Optional[str] = None,
     container_args_uses_after: Optional[str] = None,
+    monitoring=False,
```
Comment (JoanFM) [inline jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py:29]: proper type hint

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py [hunk 2]
```diff
@@ -89,16 +93,29 @@ def get_deployment_yamls(
...
+                'port': port,
+                'type': 'ClusterIP',
+                'port_monitoring': port_monitoring,
             },
-        ),
-        kubernetes_tools.get_yaml(
+        )
+
+        service_monitor_yaml = kubernetes_tools.get_yaml(
+            'service_monitor',
```
Comment (JoanFM) [inline jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py:110]: what is this service?

Comment (samsja) [reply jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py:110]: The ServiceMonitor is not a kubernetes service but  a prometheus custom object. It allows prometheus to automatically discover new service to monitor

Comment (JoanFM) [reply jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py:110]: but then the user must have some Operator or CRD installed?

**Comment by samsja** [comment]
> Don't merge this pr now I am preparing the k8s documentation on monitoring in a seperate PR dedicated to monitoring documentation. I want to be sure that what I am saying in the documentation on k8s is working before merging this one.
> 
> So I will merge both of them when the documentation one is ready

Now it is ready @JoanFM @jacobowitz

**Review by JoanFM — APPROVED** [review]

**Pull request merged**